### PR TITLE
[Path] Speed up bin generation when there are many files

### LINKED ISSUE #2846: Big gems from git hang during every install doing... something

### PR DESCRIPTION
**Pull request by segiddins**
Closes #2846, I presume.

\c @indirect

### CODE REVIEW

File: lib/bundler/source/path.rb [hunk 1]
```diff
@@ -186,6 +186,7 @@ def generate_bin(spec, disable_extensions = false)
         # Some gem authors put absolute paths in their gemspec
         # and we have to save them from themselves
         spec.files = spec.files.map do |p|
+          next p unless p =~ /\A#{Pathname::SEPARATOR_PAT}/
```
Comment (indirect) [inline lib/bundler/source/path.rb]: `SEPARATOR_PAT`? really?

Comment (segiddins) [inline lib/bundler/source/path.rb]: this is how `Pathname#relative?` is implemented

Comment (indirect) [inline lib/bundler/source/path.rb]: lolsob, alright

**Comment by indirect** [comment]
@homu r+ rollup

**Pull request merged**